src/e2ee.rs

```diff
@@ -51,23 +51,40 @@ impl EncryptHelper {
     }
 
     /// Determines if we can and should encrypt.
+    ///
+    /// For encryption to be enabled, `e2ee_guaranteed` should be true, or strictly more than a half
+    /// of peerstates, including self, should prefer encryption.
```

Comment:
Currently if self does not prefer encrypt the majority could still prefer encrypt and thus we it encrypts anyway.  I assume this is what you wanted, but that's not what I expected from reading this.  Perhaps call out that self does not have a stronger vote than any other so that this is on purpose?